presto-main/src/main/java/com/facebook/presto/dispatcher/LocalDispatchQuery.java

```diff
@@ -92,7 +95,7 @@ public LocalDispatchQuery(
         this.querySubmitter = requireNonNull(querySubmitter, "querySubmitter is null");
         this.retry = retry;
         this.queryPrerequisites = requireNonNull(queryPrerequisites, "queryPrerequisites is null");
-
+        this.warningCollector = requireNonNull(warningCollector , "warningCollector is null");
```

Comment:
the StateMachine stores the `WarningCollector` - so you can just do `stateMachine.getWarningCollector()` directly on line 130 and that will avoid a lot of the API changes

Reply:
✅